proposals/nnnn-button_subscription_response_from_hmi.md

```diff
@@ -0,0 +1,138 @@
+            <element name="OUT_OF_MEMORY"/>
+            <element name="TOO_MANY_PENDING_REQUESTS"/>
+            <element name="APPLICATION_NOT_REGISTERED"/>
+            <element name="GENERIC_ERROR"/>
+            <element name="UNSUPPORTED_RESOURCE" />
+            <element name="IGNORED" />
+            <element name="REJECTED"/>
+        </param>
+        
+        <param name="info" type="String" maxlength="1000" mandatory="false" platform="documentation">
+            <description>Provides additional human readable info regarding the result.</description>
+        </param>
+    </function>
+    
+    <function name="UnsubscribeButton" messagetype="request">
+        <description>Unsubscribes from built-in HMI buttons.</description>
+        
+        <param name="buttonName" type="ButtonName" mandatory="true">
+            <description>Name of the button to unsubscribe.</description>
+        </param>
+    </function>
+    
+    <function name="UnsubscribeButton" messagetype="response">
+        <param name="success" type="Boolean" platform="documentation" mandatory="true">
+            <description> true if successful; false, if failed </description>
+        </param>
+        
+        <param name="resultCode" type="Result" platform="documentation" mandatory="true">
+            <description>See Result</description>
+            <element name="SUCCESS"/>
+            <element name="INVALID_DATA"/>
+            <element name="OUT_OF_MEMORY"/>
+            <element name="TOO_MANY_PENDING_REQUESTS"/>
+            <element name="APPLICATION_NOT_REGISTERED"/>
+            <element name="GENERIC_ERROR"/>
+            <element name="UNSUPPORTED_RESOURCE" />
+            <element name="IGNORED" />
+            <element name="REJECTED"/>
+        </param>
+        
+        <param name="info" type="String" maxlength="1000" mandatory="false" platform="documentation">
+            <description>Provides additional human readable info regarding the result.</description>
+        </param>
+    </function>
+
+```
+
+### Storing subscriptions
+
+SDL should store the list of applications that are subscribed to a certain button. 
+According to existing API, HMI does not know anything about applications that are subscribed to a button.
+
+When application1 subscribes to a new button, SDL should send `SubscribeButton` request to HMI. 
+
+When application2 subscribes to **the same** button, SDL should **not** send anything to HMI, and store subscription internally.
+
+When any application subscribes to a **new** button, SDL should send `SubscribeButton` request to HMI. 
+
+When the last application subscribed to certain button unsubscribes SDL should send `UnsubscribeButton` request to HMI. 
+
+When the last application subscribed to certain button unregiteres SDL should send `UnsubscribeButton` request to HMI. 
+
+When HMI send OnButtonPress notification, SDL should initiale sending onButtonPress notification to all subscribed applications.
```

Comment:
1. When HMI send a `OnButtonEvent` or `OnButtonPress` notification, not just `OnButtonEvent`
2. shall SDL send the notification to all subscribed applications? Correct me if I'm wrong, I think the current behavior is send the notification to the corresponding active app.